README.md

```diff
@@ -0,0 +1,357 @@
+the remote ACL administration. For this queue, each administration command is received in a different routing key, having
+the parameters of the command serialized in the message payload. Parameters are serialized as lists of space-separated strings
+The following table shows the current allowed administration commands and their meaning and parameters:
+
+| Command             | Parameters                          | Description                                              |
+| ------------------- |:----------------------------------- |:-------------------------------------------------------- |
+| add                 | <user> <permission> <topic>         | Adds a new ACL entry with the given parameters           |
+| clear               |                                     | Remove all the ACL contents from the DB.                 |
+| save                | filename                            | Saves the contents of the ACL to the given file.         |
+| refresh             |                                     | Sends the current ACL content to the notifications topic |
+
+Some of the commands make use of a notification topic, that can be used to synchronize administration and monitoring
+systems with the contents of the ACL. This topic defaults to `notifications`.
+
+The name of the exchange is taken in runtime from the `exchange` environment variable picked up from the application
+descriptor, `rabbitmq_topic_acl.app`.
+
+An administrative tool has been developed in a separate repository to ease the administration tasks. This administrative
+tool makes use of this AMQP Administration API for all the administration tasks. It can be found in
+[this repository](https://github.com/dmoranj/rabbit-acl-tool). Refer to that repository documentation for
+further information.
+
+#### ACL Enforcing modules
+
+The `aclenforce` OTP generic server encapsulates the authorizing logic of the plugin. It exposes a single method, expose
+in the `aclenforce:authorize(User, Topic, Permission)` function for convenience. This function reads the ACL entries for
+the user from the DB, looking for the list of permissions the user have for the selected topic. If the list of permissions
+contain the permission passed as a parameter to the `authorize/3` function, the access is granted. The access is denied
+otherwise.
+
+#### Authorization plugin
+
+The authorization plugin itself is implemented as a RabbitMQ Interceptor Module. Two events are intercepted: `basic.publish`
+(emitted whenever a client publishes a message to an exchange) and `queue.bind` (emitted whenever a client tries to bind
+to a queue). Those handlers make use of the  internal function `authorize/5`, that makes use of the ACL Enforcing modules
+to determine whether it should accept the operation or not. In case the operation is not accepted, a protocol error is
+risen.
+
+In order to ease the administration tasks, the plugin allows a special administrator user, that can bypass all the
```

Comment:
I choose "guest" because is the only user created in a fresh RabbitMQ installation, and an existent user is needed here for the RabbitMQ to be started. I will add an issue to let this user be configurable (and add a warning here for the administrators to change it). NTC